internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -195,7 +199,25 @@ func (c *cloudeventProcessor) validateSignature(event *cloudevent.CloudEvent[jso
 		return false, fmt.Errorf("failed to unmarshal public key: %w", err)
 	}
 	recoveredAddress := crypto.PubkeyToAddress(*pubKey)
-	return common.HexToAddress(strings.TrimSpace(attestor)) == recoveredAddress, nil
+	if source != recoveredAddress {
+		return c.verifyERC1271Signature(signature, msgHash, source)
+	}
+
+	return true, nil
+}
+
+func (c *cloudeventProcessor) verifyERC1271Signature(signature []byte, msgHash []byte, source common.Address) (bool, error) {
+	contract, err := web3.NewErc1271(source, c.ethClient)
+	if err != nil {
+		return false, fmt.Errorf("failed to connect to address: %s: %w", source, err)
+	}
+
+	result, err := contract.IsValidSignature(nil, [32]byte(msgHash), signature)
+	if err != nil {
+		return false, fmt.Errorf("failed to validate signature with contract: %w", err)
+	}
+
+	return result != erc1271magicValue, nil
```

Comment:
I think the magic bytes mean that the signature is valid
```suggestion
	return result == erc1271magicValue, nil
```